third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -184,32 +186,31 @@ bool dds_device::impl::init()
                             LOG_ERROR( "Got VIDEO_STREAM_PROFILES with no data" );
                             break;
                         }
-
-                        for( size_t i = 0; i < video_stream_profiles->num_of_profiles; ++i )
+                        if( _streams.size() >= _expected_num_of_streams )
```

Comment:
Need to add video_stream_profiles->num_of_profiles.
And Error only if greater than, when equals need to advance state machine

Reply:
We're about to add a stream.
If we already have what we expected (or greater) then we throw an error.
What do the number of profiles have to do with it?